app/models/cart.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# This class validates the presence and format of the keys we need to calculate discounts as expected.
+# Calutates the discount price and returns the hash with the discounted_price key for each item element and the key total_price.
+class Cart
+  include ActiveModel::Validations
+  # This hash has functions as key to find the appropriate discount rate.
+  # Depending on the state of the cart (e.g how many items there are), we will pick a different discount.
+  DISCOUNT_PROGRESSION = {
+    ->(count) { count <= 1 } => 0.0,
+    ->(count) { count == 2 } => 0.05,
+    ->(count) { count == 3 } => 0.10,
+    ->(count) { count == 4 } => 0.20,
+    ->(count) { count >= 5 } => 0.25
+  }
+
+  validate :validate_line_items
+
+  def initialize(line_items = [])
+    @line_items = line_items.map(&:to_h)
+  end
+
+  def calculate_cart_discount
+    items = calculate_discounts
+
+    {
+      line_items: items,
+      total_price: items.map { |item| item[:discounted_price] }.sum.round(2)
+    }
+  end
+
+  private
+
+  def validate_line_items
```

Comment:
@victorfeijo 
Yes, absolutely. 

Once I reviewed this solution before opening a PR, I realized I could and should have used a `LineItem` class to handle line item stuff. So could not agree with you more; it makes total sense. I even wrote on the PR description _"We could have created a class to encapsulate the line item..."_ (A few considerations section). 

About validations: truly, this usually depends on the customer's needs. All you said make total sense 🙂

(I choose to do all these validations because this is an interview process, and I prefer to do it more than less since part of the process is I show you more as I can 😅)